src/storage/check.rs

```diff
@@ -1,118 +1,49 @@
 
-use std::marker::PhantomData;
-use std::ops::{Deref, DerefMut, Not};
+use std::ops::{Deref, Not};
 
 use hibitset::BitSet;
 
 use storage::{AntiStorage, MaskedStorage};
-use world::EntityIndex;
-use {Component, DistinctStorage, Index, Join, Storage, UnprotectedStorage};
+use {Component, DistinctStorage, Index, Join, Storage};
 
-/// A storage type that iterates entities that have
-/// a particular component type, but does not return the
-/// component.
-pub struct CheckStorage<'a, T, D> {
+/// Allows iterating over a storage without borrowing the storage itself.
+pub struct CheckStorage {
```

Comment:
@OvermindDL1 I'm not entirely sure? It would probably be faster if the compiler didn't optimize anything out, but in this case it probably is, so this is more for just ease of use and less trying to slide around the borrow checker.